src/odemis/gui/cont/milling.py

```diff
@@ -0,0 +1,241 @@
+        self._panel.txt_lcl_milling_est_time.Show()
+
+        # Update the milling status text
+        try:
+            future.result()
+            milling_status_txt = "Milling completed."
+        except CancelledError:
+            milling_status_txt = "Milling cancelled."
+
+        self._panel.txt_lcl_milling_est_time.SetLabel(milling_status_txt)
+
+    @wxlimit_invocation(1)  # max 1/s
+    def _update_milling_time(self):
+        """
+        Updates the estimated time
+        required for millings
+        """
+        try:
+            millings = millmng.load_config(MILLING_SETTINGS_PATH)
+        except Exception as ex:
+            logging.info("Failed to load milling settings from %s: %s", MILLING_SETTINGS_PATH, ex)
+            return
+
+        aligner = self._tab_data.main.aligner
+        ion_beam = self._tab_data.main.ion_beam
+        sed = self._tab_data.main.sed
+        stage = self._tab_data.main.stage
+
+        # filter the features that have status active
+        sites = [f for f in self._tab_data.main.features.value if f.status.value == FEATURE_ACTIVE]
+
+        feature_post_status = FEATURE_ROUGH_MILLED
+
+        acq_streams = self._tab_data.acquisitionStreams.value
+
+        millings_time = millmng.estimate_milling_time(millings, sites, feature_post_status, acq_streams,
+                                                      ion_beam, sed, stage, aligner)
+
+        # return
+
+        millings_time = math.ceil(millings_time)
+        # display the time on the GUI
+        txt = u"Estimated time: {}.".format(units.readable_time(millings_time, full=False))
+        self._panel.txt_lcl_milling_est_time.SetLabel(txt)
+
+        return
+
+    def _cancel_milling_series(self, _):
+        """
+        called when the button "Cancel" is pressed
+        """
+        logging.debug("Cancelling milling.")
+        self._mill_future.cancel()
+
+    @call_in_wx_main
+    def _on_features(self, features):
+        """
+        Updates milling time and availability of the mill button when there's an update on the features
+        """
+        self._update_milling_time()
+
+        # In case there is a new feature, also listen when its status changes
+        # (no effect on the features we already listen too)
+        for f in self._tab_data.main.features.value:
+            f.status.subscribe(self._on_features)
+
+        self._update_mill_btn()
+
+    def _on_acquisition(self, is_acquiring: bool):
+        """
+        Called when is_acquiring changes
+        Enable/Disable mill button
+        """
+        self._update_mill_btn()
+
+    @call_in_wx_main
+    def _update_mill_btn(self):
+        """
+        Enable/disable mill button depending on the state of the GUI
+        """
+        # filter the features that have status active
+        sites = [f for f in self._tab_data.main.features.value if f.status.value == FEATURE_ACTIVE]
+        # enable or disable the mill button
```

Comment:
This comment is a little bit too much stating the facts. I'd more explain "in english" the rule: "milling button is enabled if and only if there is at least one site, and no acquisition is active".